### ISSUE TITLE
Sentry block forever?

### ISSUE THREAD

**Issue by galactic16** [LevPasha/Instagram-API-python#350]
(First of all thank you for everything @LevPasha your API has been a lifesaver for me from the first day I started using it.)

`Request return 400 error!
{'status': 'fail', 'message': 'Sorry, there was a problem with your request.', 'error_type': 'sentry_block'}`

During the past month I got stupidly sentry-block multiple times, actually three times already, but every time after a bit of wait (~ 12 hours) I was able to re-run my script. 
It happened to me a fourth time yesterday and after a wait of 24 hours I'm still blocked. Do you think this time it will last forever and I'm definitely banned of using the API? Is that even possible or should I be more patient and it will eventually work again?

**Comment by Siralland01** [comment]
I am getting the same error ^^

**Comment by Siralland01** [comment]
It seems like Instagram has attempted to block the bots. This error usually occurs when instagram blocks an IP but it seems like everyone is having this problem

**Comment by galactic16** [comment]
What do you mean every one is having this problem? Only people using this Python API or every body that are botting?

**Comment by desecho** [comment]
See this issue - https://github.com/LevPasha/Instagram-API-python/issues/349, it is connected to this one. I've added a PR which at least helps to fail early and see that you are blocked clearer. Not sure if that is useful in your situation. And I added a comment with my attempt to avoid this.

**Comment by galactic16** [comment]
I actually didn't have any trouble seeing that I was blocked, the 400 error is pretty clear, a more useful functionality would be to be warned before getting blocked in order to prevent it, but I know it's not that simple since we're not even sure why we are blocked...

**Comment by Instaerror** [comment]
Which service provider are your running your bot with?

**Comment by galactic16** [comment]
@Instaerror  I run my script on a dedicated server with a given list of IPs that I can attach to the server (I use one IP per IG account and I loop through them, making my server changing IP regularly). Why?

**Comment by Instaerror** [comment]
http://www.e-chat.co/room/265618 ill be there for 10 mins!

**Comment by torlakfaruk** [comment]
@galactic16 @Instaerror could you solve this?

**Comment by akuma34** [comment]
I would suggest modifying the API to use the session, excessive use of login will anyway eventually block the account. My conclusion is that
1. I had api working for account accountA on ip ipA , this combination is still working
2. then i go and launch many server in AWS( Amazon) and Azure( Microsoft) cloud. now none of the new/old instagram account works on these new IP

but all the new/old instagram account works on my old/Original IP machine that is my residential IP in USA.

**Comment by galactic16** [comment]
@torlakfaruk No, I think that using this API from a cloud service is no longer possible.
@akuma34 You're right, I tried to run my script from my residential IP and it worked, I didn't try a second time yet though

**Comment by torlakfaruk** [comment]
Look like all the popular host service ips banned. Amazon, Azure, Google, Digitalocean ect. You should use different servers.

**Comment by desecho** [comment]
I can confirm what @galactic16 and @torlakfaruk said. Use a more conspicuous server.

**Comment by akuma34** [comment]
ok i did small experiments
1. Server from my university in US are working with this API
2. Server from my small private employer are working with this API
3. Server from AWS/Azure/GCP are not working

Now there is some inclination that they had blocked the cloud servers.

**Comment by akuma34** [comment]
Regardless, i still request to enable session reuse into this API ( i will ask my team to do it, in this week). in Any care local vs cloud server login for each cmd is not recommended and supposed to raise the flag

**Comment by desecho** [comment]
@akuma34 I implemented sessions myself for this API, seems fine like that.

**Comment by desecho** [comment]
@akuma34 You should be able to find it here - https://github.com/desecho/ighelper/blob/master/ighelper/instagram.py

**Comment by akuma34** [comment]
@desecho Thanks, this is great, now we can avoid re-login on each request.

**Comment by haliliceylan** [comment]
Hi, I have been troubling the same problem for the last 2 weeks. Although I waited, nothing changed.

I tried to change user agent but api give different error to me. ı think this error is about the ig_sig_key. how can i create this sig_key. is there any avaliable different solution ?

when i try to different ip and machine, there is no error everything is OK. However i need to a solution without changing ip.

**Comment by denizaydemir** [comment]
@haliliceylan You need to extract `IG_SIG_KEY` from memory of the device. Since different versions of app uses different APIs, you need frequent updates to both your mappings(pojo etc...) and endpoints. 
`IG_SIG_KEY = "1f189baa354be655a0153376ddb5e1a4e9101c6490cc90108006ea10bbe1c59a"; //44.0.0.9.93 android`
Here's a private key which uses current API.
